unified_planning/model/fnode.py

```diff
@@ -40,6 +41,14 @@ def __init__(self, content: FNodeContent, node_id: int):
     def __hash__(self) -> int:
         return self._node_id
 
+    ##ADDED###
+    def __deepcopy__(self, memo):
```

Comment:
Same as above, FNodes are immutable data structures and are normalized in an Environment. No need to deep-copy them